src/main/scala/expressd/ExpressD.scala

```diff
@@ -920,14 +937,16 @@ object ExpressD {
             }
 
             if (shouldUpdateAllParams) {
-              errorIndices1.add(if (errorIndex1 == null) { errorIndices1.get(0) } else { errorIndex1 })
-              errorIndices2.add(if (errorIndex2 == null) { errorIndices2.get(0) } else { errorIndex2 })
+              errorIndices1.add(if (errorIndex1 == null) { if (errorIndices1.size > 0) { errorIndices1.get(0) } else { errorIndex1 } } else { errorIndex1 })
+              errorIndices2.add(if (errorIndex2 == null) { if (errorIndices2.size > 0) { errorIndices2.get(0) } else { errorIndex2 } } else { errorIndex2 })
             }
 
             // Sum current likelihood with values in errors likelihood table at MAX_READ_LENGTH indices.
             var errLikelihood1 = LOG_1
             if (i > 0 && errorIndex1 == null) {
               errLikelihood1 = refErrLikelihood1
+            } else if ( errorIndex1 == null ) {
+              //TODO: what to do here?  was throwing NPE here -aday
```

Comment:
I'm not really sure what it means for the error indices to be null, and I don't have time to look deeply into the code right now unfortunately. @harveyfeng can you take a look at this?